stdlib/asm/word.masm

```diff
@@ -0,0 +1,190 @@
+#! Reverses order of first four elements on stack
+#!
+#! Input: [a, b, c, d, ...]
+#! Output: [d, c, b, a, ...]
+export.reverse
+    swap
+    movup.2
+    movup.3
+end
+
+#! Returns a boolean indicating whether the input word is an empty word.
+#!
+#! Inputs: [INPUT_WORD]
+#! Outputs: [is_empty_word, INPUT_WORD]
+#!
+#! - INPUT_WORD is the word whose emptiness is to be determined.
+#! - is_empty_word is a boolean indicating whether INPUT_WORD is empty.
+#!
+#! Cycles: 11
+export.is_empty
+    repeat.4
+        dup.3 eq.0
+    end
+    and and and
+end
+
+# COMPARISON OPERATIONS
+# -------------------------------------------------------------------------------------------------
+
+#! Arranges the given words such that the corresponding elements are next to each other.
+#!
+#! Inputs:  [WORD1, WORD2]
+#! Outputs: [word2_3, word1_3, word2_2, word1_2, word2_1, word1_1, word2_0, word1_0]
+proc.arrange_words_adjacent
+    # => [1_3, 1_2, 1_1, 1_0, 2_3, 2_2, 2_1, 2_0]
+
+    movup.3 movup.7
+    # => [2_0, 1_0, 1_3, 1_2, 1_1, 2_3, 2_2, 2_1]
+
+    movup.4 movup.7
+    # => [2_1, 1_1, 2_0, 1_0, 1_3, 1_2, 2_3, 2_2]
+
+    movup.5 movup.7
+    # => [2_2, 1_2, 2_1, 1_1, 2_0, 1_0, 1_3, 2_3]
+
+    movup.6 movup.7
+    # => [2_3, 1_3, 2_2, 1_2, 2_1, 1_1, 2_0, 1_0]
+end
+
+#! Returns true if LHS is strictly greater than RHS, false otherwise.
+#!
+#! This compares words using the same ordering as Merkle tree key comparisons.
+#!
+#! The implementation avoids branching for performance reasons.
+#!
+#! For reference, this is equivalent to the following Rust function:
+#!
+#! fn is_key_greater(key1: Word, key2: Word) -> bool {
+#!     let mut result = false;
+#!     let mut cont = true;
+#!
+#!     for i in (0..4).rev() {
+#!         let gt = key1[i].as_int() > key2[i].as_int();
+#!         let eq = key1[i].as_int() == key2[i].as_int();
+#!         result |= gt & cont;
+#!         cont &= eq;
+#!     }
+#!
+#!     result
+#! }
+#!
+#! Inputs:  [LHS, RHS]
+#! Outputs: [is_lhs_greater]
+export.gt
```

Comment:
For completeness, I would also add `gte` and `lte` procedures (which should be easy to add given the ones already existing here).